src/screen/menu.hpp

```diff
@@ -88,6 +89,13 @@ class Menu : public Screen {
             0,
             MenuItem::ItemType::VALUE_BINARY,
             true,
+        }, {
+            "High power:";
```

Comment:
Ooh, good catch, thank you. But honestly, if we agree that it is better to automatically detect power levels (which I think also, I just hadn't figured out how so I was forcing it with a menu), then all these code changes would go away. The only change required would be to heating.hpp and screen/main.hpp (for the power bar). Settings.hpp and screen/menu.hpp could be unchanged from your version.

But, if we intend to add RTPico tip function, then we cannot auto-detect this, as they have the same resistance as RTM (I think that's correct?). So we would have to have a setting for RTM vs. RTP to change the PIDs used, if new PIDs are developed for all three tips.

Reply:
Yes, detecting RTM vs RTU is easy, but RTM vs RTP is also possible, RTP tips are very small and my idea is to somehow provide "heating speed" parameter from PID regulation loop, this can easily identify TIP type. but currently it is only idea, I have no practical researches about this ,-)

Another idea is to select right tip from menu and all these mechanisms can only stop heating if some detection will say something else.

Also RTM and RTP tips uses same parameters (40W/12V/2ohm) so there is no problem.. ok one, RTP tips require little different PID constants.

Reply:
I would much much prefer to be able to switch between RTU and RTM and not have to worry about going into menus to make changes. But I also don't intend to use RTP tips very much, if at all, it's just not the type of soldering I do. But, I think, if there will be different PIDs for RTP and RTM, then that could be a boolean setting in the menu. I suspect it's pretty unlikely that someone would be switching constantly between RTU, RTM, and RTP all at the same time, so at least if one of the parts can be automatic, that would probably make life easier for some folks.

And then if you figure out a way to automate detection of the RTP/RTM, then that can be added, too. But that stuff is way above my skill level, haha.